module2/gift-certificate-system/persistence/src/main/java/com/epam/esm/model/GiftCertificate.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.esm.model;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor()
```

Comment:
Do you need @AllArgsConstructor for this class? And also @Data annotation provides default constructor in common case, so you can remove @NoArgsConstructor as well.

Reply:
I had a problem with deserialization in Jackson. On StackOverflow I found a description of this bug and its solution

https://github.com/rzwitserloot/lombok/issues/1563#issuecomment-363460962